services/powchain/src/main/java/tech/pegasys/teku/services/powchain/execution/client/Web3JExecutionEngineClient.java

```diff
@@ -44,23 +47,22 @@ public Web3JExecutionEngineClient(String eth1Endpoint, Optional<String> maybeEeE
   }
 
   @Override
-  public SafeFuture<Response<PreparePayloadResponse>> preparePayload(
-      PreparePayloadRequest request) {
-    Request<?, PreparePayloadWeb3jResponse> web3jRequest =
+  public SafeFuture<UInt64> preparePayload(PreparePayloadRequest request) {
+    Request<?, UInt64Web3jResponse> web3jRequest =
         new Request<>(
             "engine_preparePayload",
             Collections.singletonList(request),
             eeWeb3jService,
-            PreparePayloadWeb3jResponse.class);
-    return doRequest(web3jRequest);
+            UInt64Web3jResponse.class);
+    return doRequest1(web3jRequest).thenApply(UInt64Web3jResponse::getUInt);
```

Comment:
(nit) `doRequest1` name is odd, maybe we could find something better when we will have the final version of the APIs if this is still required.

Reply:
Oh, yes, thanks! That was a working method name :)

Reply:
That method was finally removed